Bluetooth: Mesh: Remove net_idx params when with app key

LingaoM opened the pull request:
According Mesh Profile 1.0.1. A application key shall
binding single network key. And Device key shall bind all
network key, and dev key only known by cfg_cli and node self,
only used by cfg_cli & cfg_srv.

Fixes: #21088

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

LingaoM committed:
Bluetooth: Mesh: Remove net_idx params when with app key

According Mesh Profile 1.0.1. A application key shall
binding single network key. And Device key shall bind all
network key, and dev key only known by cfg_cli and node self,
only used by cfg_cli & cfg_srv.

Fixes: #21088

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

jhedberg commented:
If we merge this we should add a note about the Health Client breaking API changes to the release notes. @carlescufi FYI

carlescufi commented:
> If we merge this we should add a note about the Health Client breaking API changes to the release notes. @carlescufi FYI

Indeed, @LingaoMeng could you  add a note in `doc/releases/release-notes-2.3.rst`, under "Stable API changes"?

jhedberg commented on doc/releases/release-notes-2.3.rst:
```diff
@@ -51,6 +51,11 @@ Removed APIs in this release
 
+* Bluetooth Mesh
+
+  * Health client model API has been reworded to deteted `net_idx` parameter.
+  Since we no longer need this parameter, it can be derived from app_idx.
```
I'd suggest changing this to: "The `net_idx` parameter has been removed from the Health Client model APIs since it can be derived (by the stack) from the `app_idx` parameter".

Also, I think the preference is to have release notes changes in independent commits. @carlescufi is that correct?

jhedberg commented:
@LingaoMeng thanks for splitting up the commit, but the release notes commit should be prefixed with "doc: relnotes: ..." instead of "Bluetooth: Mesh: ..."

trond-snekvik commented on subsys/bluetooth/mesh/access.c:
```diff
@@ -688,6 +688,19 @@ int bt_mesh_model_send(struct bt_mesh_model *model,
+		app_key = bt_mesh_app_key_find(ctx->app_idx);
+		if (!app_key) {
+			BT_ERR("Unable find binded netkey (app_idx: 0x%04x)",
+					ctx->app_idx);
```
The error here is technically "Unknown app_idx 0x%04x", I think this would be more helpful for debugging.

trond-snekvik reviewed (changes requested):
Good change. 

I think this should be documented in the msg_ctx struct. e.g. "NetKey Index of the subnet to send the message on when using a device key."

LingaoM committed:
doc: relnotes: Remove net_idx params in health_cli API

The net_idx parameter has been removed from the Health Client
model APIs since it can be derived (by the stack) from the
app_idx parameter

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>
